### ISSUE TITLE
fix Loadable state names to be parallel

### ISSUE THREAD

**Issue by ianstormtaylor** [facebookexperimental/Recoil#878]
Right now the `Loadable` state names are `loading`, `hasError`, and `hasValue`. But the extra "has" is weird because why is it not also then `isLoading`? It's a small fix, but would eliminate an extra bit of inconsistency in the API, and make reading the state slightly terser too.

There are a number of ways to do it, but I think the simplest change is:

```
loading
error
value
```

You could go the other way and add "is" (but I don't see the need to add extra characters and extra camel-case, especially when the prefix doesn't match up which just makes it slightly harder to remember):

```
isLoading
hasError
hasValue
```

Or if you wanted to mirror the existing [Promise state terms](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) to not re-invent:

```
pending
fulfilled
rejected
```

**Comment by drarmstr** [comment]
I completely agree that the names could be improved.  However, they are already part of the API and the incremental improvement in usability would have to offset the issue of backward compatibility.  Maybe something to consider for a major release.
